### ISSUE TITLE
Update example code due to deprecations in RxJS

### ISSUE THREAD

**Issue by DoktorZ** [angular/angular#44708]
### Description

In documentation in example codes there are many usages of RxJS.
Several functions are going to be deprecated or already deprecated and will be removed in next major release.

For example in developer guide for HTTP Client, interceptors part ([Logging request and response pairs](https://angular.io/guide/http#logging-request-and-response-pairs)) the following code-snippet with tap-function uses deprecated form of the tap:

```
...
tap(
  // Succeeds when there is a response; ignore other events
  event => ok = event instanceof HttpResponse ? 'succeeded' : '',
  // Operation failed; error is an HttpErrorResponse
  error => ok = 'failed'
),
...
```

The deprecation note for the function used: "Instead of passing separate callback arguments, use an observer argument. Signatures taking separate callback arguments will be removed in v8. Details: https://rxjs.dev/deprecations/subscribe-arguments"

See more for details: https://rxjs.dev/api/operators/tap

### Proposed solution

The replacement of the provided snippet by the following:

```
...
tap({
  next: event => ok = event instanceof HttpResponse ? 'succeeded' : '',
  error: error => ok = 'failed'
}),
...
```

### Alternatives considered

No alternatives.

**Comment by yousafnawaz** [comment]
Hi @JoostK, If its not assigned to anyone. I will be providing PR to this.

**Comment by JoostK** [comment]
@yousafnawaz go for it, thank you!

**Comment by yousafnawaz** [comment]
Hi @JoostK, header of commit will be "refactor(http): update deprecated form of the tap" ?

**Comment by yousafnawaz** [comment]
> You could also use the `docs` prefix. I'd suggest to change `the tap` into `the tap operator`, as without _operator_ that sentence looks unfinished.
> 
> Please also include a `Closes #44708` comment in a new paragraph such that this issue is automatically closed when the commit is merged.

"docs(examples): update deprecated form of the tap operator" looks fine now?

**Comment by JoostK** [comment]
Yep
